cache/test_conditional.py

```diff
@@ -461,11 +455,95 @@ def test_cachable_headers(self, name, header):
         )
         self.assertIn(header, client.last_response.headers.items())
         self.assertIn(("date", date), client.last_response.headers.items())
 
+    @marks.Parameterize.expand(
+        [
+            marks.Param(
+                name="vary",
+                header=("vary", "accept-language"),
+                expected_status_code="304",
+                date=HttpMessage.date_time_string(),
+                disable_deproxy_auto_parser=False,
+            ),
+            marks.Param(
+                name="content-location",
+                header=("content-location", "/documents/page.html"),
+                expected_status_code="304",
+                date=HttpMessage.date_time_string(),
+                disable_deproxy_auto_parser=False,
+            ),
+            marks.Param(
+                name="expires",
+                header=("expires", "Thu, 01 Dec 2102 16:00:00 GMT"),
+                expected_status_code="304",
+                date=HttpMessage.date_time_string(),
+                disable_deproxy_auto_parser=False,
+            ),
+            marks.Param(
+                name="date_invalid",
+                header=("expires", "Thu, 01 Dec 2102 16:00:00 GMT"),
+                expected_status_code="200",
+                date="Thu, 01 Dec 2102 16:00:00 GMT111",
+                disable_deproxy_auto_parser=True,
+            ),
+            marks.Param(
+                name="expires_and_date_invalid",
+                header=("expires", "Thu, 01 Dec 2102 16:00:00 GMT111"),
+                expected_status_code="200",
+                date="Thu, 01 Dec 2102 16:00:00 GMT111",
+                disable_deproxy_auto_parser=True,
+            ),
+            marks.Param(
+                name="cache-control",
+                header=("cache-control", "public"),
+                expected_status_code="304",
+                date=HttpMessage.date_time_string(),
+                disable_deproxy_auto_parser=False,
+            ),
+            marks.Param(
+                name="etag",
+                header=("etag", '"etag"'),
+                expected_status_code="304",
+                date=HttpMessage.date_time_string(),
+                disable_deproxy_auto_parser=False,
+            ),
+        ]
+    )
+    def test_cachable_headers(
+        self, name, header, expected_status_code, date, disable_deproxy_auto_parser
+    ):
+        self.__test_cachable_headers(
+            header, expected_status_code, date, disable_deproxy_auto_parser
+        )
+
+    @marks.Parameterize.expand(
+        [
+            marks.Param(
+                name="expires_invalid",
+                header=("expires", "Thu, 01 Dec 2102 16:00:00 GMT111"),
+                expected_status_code="200",
+                date=HttpMessage.date_time_string(),
+                disable_deproxy_auto_parser=False,
+            ),
+        ]
+    )
+
+    @unittest.expectedFailure
+    def test_cachable_headers(
```

Comment:
Methods with the same name, but different decorators looks confusing, we can reflect in the name the fact that this version is expected to failure.